### ISSUE TITLE
"Nginx not found" when compiling waf-3

### ISSUE THREAD

**Issue by Acharnite** [theonemule/docker-waf#10]
I've tried setting up the waf-3 a couple of times now, and each time i get a "nginx not found"

waf-2 works flawlessly.

**Comment by OnePython** [comment]
I have the same problem, don't know how to solve it?

**Comment by Acharnite** [comment]
Personally I haven't had time to look into it yet.
Hoping I can find some time for it this weekend, and hopefully find a solution :)

**Comment by NicklasXYZ** [comment]
It is probably not the best way to do it, but I found a quick fix to this problem.

In the `waf-3/Dockerfile` do the following:
1. Replace `CMD nginx -g 'daemon off'` with `CMD /usr/local/nginx/sbin/nginx -g 'daemon off'`. The reason for this is that when running `make install` in the Docker file the nginx executable is not placed in the `/bin` directory, but is instead installed to the directory `/usr/local/nginx/sbin/nginx`.
2. Remove the `apt-get autoremove -y` from the Dockerfile. The reason for this is that the nginx dependency libxml2 is removed by the use of this command.

Conclusively, it works for me if I use the Dockerfile:

``
FROM ubuntu:18.04
RUN apt-get update && \
	apt-get install -y wget tar apt-utils autoconf automake build-essential git libcurl4-openssl-dev libgeoip-dev liblmdb-dev libpcre++-dev libtool libxml2-dev libssl-dev libyajl-dev pkgconf zlib1g-dev && \
	git clone --depth 1 -b v3/master --single-branch https://github.com/SpiderLabs/ModSecurity && \
	cd ModSecurity && \
	git submodule init && \
	git submodule update && \
	./build.sh && \
	./configure && \
	make && \
	make install && \
	cd / && \
	git clone --depth 1 https://github.com/SpiderLabs/ModSecurity-nginx.git && \
	wget http://nginx.org/download/nginx-1.18.0.tar.gz && \
	tar zxvf nginx-1.18.0.tar.gz && \
	cd nginx-1.18.0 && \
	./configure --user=root --group=root --with-debug --with-ipv6 --with-http_ssl_module  --with-compat --add-module=/ModSecurity-nginx --without-http_access_module --without-http_auth_basic_module --without-http_autoindex_module --without-http_empty_gif_module --without-http_fastcgi_module --without-http_referer_module --without-http_memcached_module --without-http_scgi_module --without-http_split_clients_module --without-http_ssi_module --without-http_uwsgi_module && \
	make && \
	make install && \
	cd / && \
	git clone https://github.com/SpiderLabs/owasp-modsecurity-crs.git /usr/src/owasp-modsecurity-crs && \
	cp -R /usr/src/owasp-modsecurity-crs/rules/ /usr/local/nginx/conf/  && \
	mv /usr/local/nginx/conf/rules/REQUEST-900-EXCLUSION-RULES-BEFORE-CRS.conf.example  /usr/local/nginx/conf/rules/REQUEST-900-EXCLUSION-RULES-BEFORE-CRS.conf && \
	mv /usr/local/nginx/conf/rules/RESPONSE-999-EXCLUSION-RULES-AFTER-CRS.conf.example  /usr/local/nginx/conf/rules/RESPONSE-999-EXCLUSION-RULES-AFTER-CRS.conf && \
	apt-get remove -y --purge apt-utils autoconf automake build-essential git pkgconf
COPY nginx.conf /usr/local/nginx/nginx.conf
COPY modsec_includes.conf /usr/local/nginx/modsec_includes.conf
COPY modsecurity.conf /usr/local/nginx/modsecurity.conf
COPY crs-setup.conf /usr/local/nginx/crs-setup.conf
CMD ./usr/local/nginx/sbin/nginx -g 'daemon off;'
EXPOSE 80
``

**Comment by jordan-commited** [comment]
brilliant! thanks

**Comment by theonemule** [comment]
Thanks for that. I incorporated the CMD option into the Dockerfile in the main repo.

**Comment by NicklasXYZ** [comment]
You're welcome :)

But I think there still might be a problem.

I actually ended up adding this repository (as a remote repository) to another forked project I have.
I changed quite a few things in the dockerfiles, but if I remember correctly I ran into another problem where the wrong `nginx.conf` file was loaded by nginx.

The `nginx.conf` file that is copied into the container is actually not used:

`COPY nginx.conf /usr/local/nginx/nginx.conf`

Instead it seems like nginx is actually using another default `nginx.conf` file located in /usr/local/nginx/conf/nginx.conf, _if I remember correctly._

The dockerfiles I eventually used (based on this repo) is located in the waf3_dep and waf3 directories in https://github.com/NicklasXYZ/Docker-NGINX-Certbot-ModSecurity
